workflow/deterministic_wrappers.go

```diff
@@ -111,6 +111,21 @@ func AwaitWithTimeout(ctx Context, timeout time.Duration, condition func() bool)
 	return internal.AwaitWithTimeout(ctx, timeout, condition)
 }
 
+// AwaitWithOptions blocks the calling thread until condition() returns true
+// or blocking time exceeds the passed timeout value.
+// Returns ok=false if timed out, and err CanceledError if the ctx is canceled.
+// The following code will block until the captured count
+// variable is set to 5, or one hour passes.
+//
+//	workflow.AwaitWithOptions(ctx, options, func() bool {
+//	  return count == 5
+//	})
+//
+// NOTE: Experimental
+func AwaitWithOptions(ctx Context, options internal.AwaitOptions, condition func() bool) (ok bool, err error) {
```

Comment:
You'll need a type alias for `AwaitOptions` in the public package here to reference the internal, otherwise users can't even call it. This may surface if/when you write integration tests.

Reply:
Is it worth adding a unit test for `AwaitWithOptions` that does the same thing as `AwaitWithTimeout` just to exercise the new function codepath in our tests?

Reply:
Doesn't necessarily do the same thing, it also allows you to add a summary. Might as well test it, but I don't have a strong opinion. I think it'd have caught that certain needed things weren't exposed here though.